File: RnCliProjectSandbox/ios/MyTextViewManager.m
#import "React/RCTViewManager.h"

@interface

RCT_EXTERN_MODULE(MyTextViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(title, NSString)
RCT_EXPORT_VIEW_PROPERTY(textAlign, NSTextAlignment)
RCT_EXPORT_VIEW_PROPERTY(color, UIColor)

@end

File: RnCliProjectSandbox/ios/MyTextViewManager.swift
import UIKit
import React

@objc(MyTextViewManager)
class MyTextViewManager: RCTViewManager {

  override func view() -> UIView! {
    MyLabel()
  }

  override func shadowView() -> RCTShadowView! {
    let shadowView = RCTTextShadowView.init(bridge: self.bridge)
    return shadowView
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

class MyLabel: UILabel {
  @objc
  func setTitle(_ value: String) {
    text = value
  }

  @objc
  func setTextAlign(_ value: NSTextAlignment) {
    textAlignment = value
  }

  @objc
  func setColor(_ value: UIColor) {
    textColor = value
  }
}
